docs/cosmology/io.rst

```diff
@@ -86,23 +167,39 @@ As a quick example, outlining how to make a
 .. code-block:: python
 
    import json
+   import os
    from astropy.cosmology import Cosmology
    from astropy.io import registry as io_registry
 
    def read_json(filename, **kwargs):
-       cosmology = ...  # read and parse file
-       return cosmology
-
-   def write_json(cosmology, file, **kwargs):
-       data = ...  # parse cosmology to dict
-       with open(file, "w") as write_file:
-            json.dump(data, write_file)
+       # read file, from path-like or file-like
+       if isinstance(filename, (str, bytes, os.PathLike)):  # pathlike
+           with open(filename, "r") as file:
+               data = file.read()
+       else:  # file-like
+           data = filename.read()
+
+       mapping = json.loads(data)  # parse json to dict
+       return Cosmology.from_format(mapping, move_to_meta=move_to_meta)
```

Comment:
`move_to_meta` is not defined...

Would be nice to ensure the code here is tested.  One suggestion is to explicit refer to the test file and to `cosmology.io` for examples.

Reply:
About ``move_to_meta`` here and in the comment above... thanks! I've improved the docs to have doctest examples. While ``move_to_meta`` isn't necessary presently, it's integral to allowing 3rd party registries where cosmology objects have e.g. precision flags.  I just need to make some tests in this PR that use ``move_to_meta``.

Reply:
I'm actually trying to remove all mention of ``cosmology.io`` in the narrative docs. The test examples now use ``Cosmology.to/from_format`` instead of ``to/from_mapping`` and point users to ``Cosmology.to/from_format.help()``. I would like to be able to show the details of ``help`` for each format without linking to the private function, but I don't think that's currently possible.